### DISCUSSION TITLE
Solving ODEs using Fast Fourier Transform

### DISCUSSION THREAD

**Discussion by Jamiewhbee** [Part-II-Computational-Physics/.github#6] [Q&A]
I'm having difficulty applying the procedure for the spectral scheme:
1. Calculate FFT of f^n
2. Use either explicit upwind scheme or Lax-Wendroff scheme to calculate FFT of f^n+1 from FFT of f^n
3. Calculate inverse FFT of f^n+1 

However, the FFT of f^n+1 will generally be complex, so should I take the real part, the absolute value, or something else before plotting?
Also (probably related to the above by the symmetry of the FT) I can't get my solution to propagate - it instead grows with time before diverging, while the peak remains fixed in space (see below).

![Spectral](https://user-images.githubusercontent.com/122116209/219785429-f2f95f28-a9f8-4a23-9a01-e848ea4ec36b.gif)

**Comment by Jamiewhbee** [comment]
I think I misinterpreted the task, since the differential equation at this stage is ordinary rather than partial, so techniques such as Euler's method and the midpoint method should work.

**Reply by NathanMagnan** [reply]
Yes, and (hint) if you choose an Euler method, choose one that is stable.
